gadget/update_test.go

```diff
@@ -573,3 +578,618 @@ func (u *updateTestSuite) TestCanUpdateVolume(c *C) {
 
 	}
 }
+
+type mockUpdater struct {
+	updateCb   func() error
+	backupCb   func() error
+	rollbackCb func() error
+}
+
+func callOrNil(f func() error) error {
+	if f != nil {
+		return f()
+	}
+	return nil
+}
+
+func (m *mockUpdater) Backup() error {
+	return callOrNil(m.backupCb)
+}
+
+func (m *mockUpdater) Rollback() error {
+	return callOrNil(m.rollbackCb)
+}
+
+func (m *mockUpdater) Update() error {
+	return callOrNil(m.updateCb)
+}
+
+func updateDataSet(c *C) (oldData gadget.GadgetData, newData gadget.GadgetData, rollbackDir string) {
+	// prepare the stage
+	bareStruct := gadget.VolumeStructure{
+		Name: "first",
+		Size: 5 * gadget.SizeMiB,
+		Content: []gadget.VolumeContent{
+			{Image: "first.img"},
+		},
+	}
+	fsStruct := gadget.VolumeStructure{
+		Name:       "second",
+		Size:       10 * gadget.SizeMiB,
+		Filesystem: "ext4",
+		Content: []gadget.VolumeContent{
+			{Source: "/second-content", Target: "/"},
+		},
+	}
+	lastStruct := gadget.VolumeStructure{
+		Name:       "third",
+		Size:       5 * gadget.SizeMiB,
+		Filesystem: "vfat",
+		Content: []gadget.VolumeContent{
+			{Source: "/third-content", Target: "/"},
+		},
+	}
+	oldInfo := &gadget.Info{
+		Volumes: map[string]gadget.Volume{
+			"foo": {
+				Bootloader: "grub",
+				Schema:     gadget.GPT,
+				Structure:  []gadget.VolumeStructure{bareStruct, fsStruct, lastStruct},
+			},
+		},
+	}
+	newInfo := &gadget.Info{
```

Comment:
Could use a small comment, e.g. "oldInfo and newInfo intentionally the same, they get updated by the caller".